core/src/main/java/nu/yona/server/subscriptions/service/PinResetRequestService.java

```diff
@@ -80,9 +82,10 @@ public void verifyPinResetConfirmationCode(UUID userId, String userProvidedConfi
 		User userEntity = userService.getUserEntityById(userId);
 		logger.info("User with mobile number '{}' and ID '{}' requested to verify the pin reset confirmation code",
 				userEntity.getMobileNumber(), userId);
-		ConfirmationCode confirmationCode = userEntity.getPinResetConfirmationCode();
-		Require.that((confirmationCode != null) && !isExpired(confirmationCode),
+		Optional<ConfirmationCode> confirmationCodeOpt = userEntity.getPinResetConfirmationCode();
+		Require.that(isValidConfirmationCode(confirmationCodeOpt),
 				() -> PinResetRequestConfirmationException.confirmationCodeNotSet(userEntity.getMobileNumber()));
+		ConfirmationCode confirmationCode = confirmationCodeOpt.get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "confirmationCodeOpt.isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS3655)

Reply:
False positive. The check on line 86 verifies that the confirmation code is present.